### PR TITLE
Remove double unescape

### PR DESCRIPTION
**Pull request by xonstone**
The call to `r.URL.Query().Get("filter")` already triggers an unescape so doing it again yields incorrect values. Resolves https://github.com/elimity-com/scim/issues/133

### CODE REVIEW

**Comment by di-wu** [comment]
Thank you for the PR @xonstone!
It is possible to add tests that cover this fix?

**Comment by xonstone** [comment]
Sure, I checked server_test.go but couldn't really find a good spot to add these. Is there already a place for this or should I start from scratch for these (eg this type of test probably needs some mocks, ...)

**Comment by xonstone** [comment]
I checked it out a bit further, I'll add something similar to `patch_add_test.go`

**Comment by xonstone** [comment]
@di-wu I've added the tests and the PR is ready to be reviewed again!

File: server_test.go [hunk 1]
```diff
@@ -117,10 +117,6 @@ func (h *testResourceHandler) Create(r *http.Request, attributes scim.ResourceAt
 }
 
 func (h *testResourceHandler) Delete(r *http.Request, id string) error {
-	if err := checkBodyNotEmpty(r); err != nil {
-		return err
-	}
-
```
Comment (di-wu) [inline server_test.go:123]: Is there a reason these are removed?

Comment (xonstone) [reply server_test.go:123]: Yes, on all the read endpoints these dont make sense since that will always be without a body

**Comment by xonstone** [comment]
@di-wu Anything else blocking this PR?

**Review by di-wu — APPROVED** [review]
LGTM

**Comment by di-wu** [comment]
> Anything else blocking this PR?

Just forgot to approve, thanks for the PR @xonstone!

**Pull request merged**